lang/ofc/Portfile

```diff
@@ -9,28 +9,39 @@ PortGroup           makefile 1.0
 # strndup
 legacysupport.newest_darwin_requires_legacy 10
 
-github.setup        CodethinkLabs ofc 1 release/
-revision            1
-categories          lang
+github.setup        CodethinkLabs ofc b69ad37be62d580eabeb081e64b43afe3c0b3384
+version             2023.02.06
+revision            0
+categories          lang fortran
 license             Apache-2
 maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
 description         Open Fortran Compiler
 long_description    Currently OFC is a Fortran front-end capable of parsing and performing semantic analysis \
                     on Fortran. We’re targeting legacy Fortran first, and can currently parse and semantically analyse \
                     most F77 and earlier, providing syntax and semantic warnings and errors.
-checksums           rmd160  3435ba4379233f0a55014aa9509d68f5979dfac7 \
-                    sha256  79bb5f9902a50c7094044def371278f8b29ce4c6b485abb6ee1b660ab2fd229c \
-                    size    196221
+checksums           rmd160  64d56a24c41295fd0f29aac144b790531e15db41 \
+                    sha256  55a66c5d9cf51843bde6fc0442a9cbc66e6c5ff700bfe52cfb840ef998803767 \
+                    size    199538
+github.tarball_from archive
 installs_libs       no
 
-# See: https://github.com/CodethinkLabs/ofc/issues/45
-patchfiles          patch-upstream.diff
-
 compiler.blacklist  *gcc-4.* {clang < 421}
 
 post-patch {
     reinplace "s|/usr/local|${prefix}|" ${worksrcpath}/Makefile
 
+    if {[string match *clang* ${configure.compiler}]} {
+        if {${os.platform} eq "darwin" && ${os.major} > 20} {
+            # unformat.c: error: variable 'row' set but not used [-Werror,-Wunused-but-set-variable]
+            # https://github.com/CodethinkLabs/ofc/issues/49
+            reinplace "s|-Werror|-Werror -Wno-error=unused-but-set-variable|g" ${worksrcpath}/Makefile
+        } elseif {${os.platform} eq "darwin" && ${os.major} == 19} {
+            # file.c: error: declaration does not declare anything [-Werror,-Wmissing-declarations]
+            # https://github.com/CodethinkLabs/ofc/issues/46
+            reinplace "s|-Werror|-Werror -Wno-error=missing-declarations|g" ${worksrcpath}/Makefile
+        }
+    }
```

Comment:
why not remove `-Werror` regardless of the OS versions - we normally don't build with that flag anyway. I don't think there is a reason to have these conditionals here.

Reply:
@reneeotten I would rather keep it, but won’t insist: if you prefer dropping -Werror unconditionally, I am fine with that either.

Reply:
I've seen Josh recently replacing `-Werror` with `-Wextra` in a few ports, so perhaps doing that unconditionally would be a suitable solution?

Reply:
@reneeotten Since I cannot test anything on older OS with GCC at the moment, let me rather just remove `-Werror` here.

Reply:
FWIW, I think -Werror was appropriate but it doesn't matter much. (-Wextra is like -Werror but "harder").